### ISSUE TITLE
RCS0034 False positive

### ISSUE THREAD

**Issue by daze99** [dotnet/roslynator#1347]
**Product and Version Used:**
Roslynator 4.8.0

**Steps to Reproduce:**
```csharp
public interface IBaseInterface<TType, TKey>
    where TType : class
    where TKey : struct, IEquatable<TKey>
{
}

public interface IInterface<TType, TKey> : IBaseInterface<TType, TKey>
    where TType : class
    where TKey : struct, IEquatable<TKey>
{
}
```

**Actual Behavior**:
The line `public interface IInterface<TType, TKey> : IBaseInterface<TType, TKey>` produces a false positive.
